src/WooCommerceCheckout.php

```diff
@@ -118,4 +122,34 @@ public static function checkConfirmationEmailField() {
     }
   }
 
+  /**
+   * Check if we are using Amazon V2 in the checkout.
+   *
+   * @return bool
+   */
+  public static function isAmazonPayV2Checkout(): bool {
+    $is_amzon_pay_active = is_plugin_active('woocommerce-gateway-amazon-payments-advanced/woocommerce-gateway-amazon-payments-advanced.php') ?: FALSE;
+    if ($is_amzon_pay_active) {
+      if(version_compare(WC_AMAZON_PAY_VERSION, '2.0', '>=') && WC()->session->get('amazon_checkout_session_id') !== null) {
```

Comment:
I am using the constant `WC_AMAZON_PAY_VERSION` provided by the plugin itself instead of getting it with `get_plugin_data()` because their plugin header doesn't adjust to WordPress standards and it was not returning the correct info: https://developer.wordpress.org/reference/functions/get_plugin_data/

Reply:
That's okay, but you should check whether the constant is defined before using it in order to prevent a PHP warning in the future in case the plugin changes its constants.

It seems like you're not actually interested in the content, only whether the session parameter is set.  Therefore, you should use `isset()` or `!empty()` for this instead of comparing against `null`.

Doing this should also allow you to simplify and shorten the code to a single `isset()`, because PHP evaluates the expression from left to right and returns a negative result for the whole `isset()` expression if any property or value in the chain is not set.

Please also note that your code is not following our coding standards in various places (here: missing space after `if`).
```suggestion
      if (defined('WC_AMAZON_PAY_VERSION') && version_compare(WC_AMAZON_PAY_VERSION, '2.0', '>=') && isset(WC()->session->get('amazon_checkout_session_id'))) {
```

Reply:
@sun "It seems like you're not actually interested in the content, only whether the session parameter is set. Therefore, you should use isset() or !empty() for this instead of comparing against null."

You are correct, I am not interested in the content, but without doing that comparison I get this error, so I have left it as it. Any advise?

`It seems like you're not actually interested in the content, only whether the session parameter is set. Therefore, you should use isset() or !empty() for this instead of comparing against null.`